nix/flake-modules/formatter.nix

```diff
@@ -0,0 +1,66 @@
+{ inputs, ... }:
+{
+  imports = [
+    inputs.treefmt-nix.flakeModule
+  ];
+  perSystem =
+    { config, pkgs, ... }:
+    {
+      # Auto formatters. This also adds a flake check to ensure that the
+      # source tree was auto formatted.
+      treefmt.config = {
```

Comment:
Is this actually the right setting? I was reading the [treefmt-nix flake module](https://github.com/numtide/treefmt-nix/blob/4446c7a6fc0775df028c5a3f6727945ba8400e64/flake-module.nix) a bit, and it seems like all the settings are just under `treefmt.*` (eg `treefmt.settings` instead of `treefmt.config.settings`).

Reply:
"it must be right otherwise it would be showing an error".

I am somewhat curious now too. I don't see any definitions for `treefmt.config` but if I renamed it to `treefmt.xxx` it clearly showed an error. So `treefmt.config` must be defined somewhere.

Reply:
I see options documented here, no mention of `treefmt.config`: https://flake.parts/options/treefmt-nix

Reply:
> it must be right otherwise it would be showing an error

Haha I was thinking this too. Well I guess if it's working then it's fine. It is odd though that `treefmt.config` doesn't seem to be defined anywhere... 

Oh maybe it's quirk of the fact that `treefmt` is defined a submodule? Like if you did `treefmt = { ... }: { /* whatever */ }` I'm guessing that might work as well.

Reply:
> Is this actually the right setting?

Yeah, this bothered me a lot, until I found the right setting in the flake-parts dock

> I see options documented here, no mention of `treefmt.config`: https://flake.parts/options/treefmt-nix
Exactly this.

Reply:
Not 100% sure I understand where we got to here.

I would recommend replacing `treefmt.config = {` with just `treefmt = {` - as per the documentation - unless there is a good reason not to (e.g. it is breaking something).

> until I found the right setting in the flake-parts dock

What is this?

Reply:
> I would recommend replacing `treefmt.config = {` with just `treefmt = {` - as per the documentation - unless there is a good reason not to (e.g. it is breaking something).

Not running in issues with just `treefmt = {}`. I tooked it from: https://github.com/srid/haskell-template/blob/master/nix/modules/formatter.nix